module-2/data-analysis-with-pandas/your-code/main.ipynb

```diff
@@ -1,13 +1,44 @@
 {
```

Comment:
- It's a good practice to include at the begining of your analysis a description of your data set including the meaning of the variables(columns). For example, this column 'NFFL Fuel Model' is a categorical variable. Setting variables as categorical facilitates the plotting process. 
- One of your best friends should be pandas-profiling. This library offer some useful methods to get an overview of your data after importing. Run on your term pip install pandas-profiling. In your notebook import pandas-profiling and apply the following to one of your dataframes. 
```Python
pandas_profiling.ProfileReport(df)
```
- One of the first thing you should do when analysing your dataset is a review of your datatypes and ensure you properly identify categorical variables(even if they are encoded) and numerical ones. 
- When you want to know missing values you don't need to run the following. Running the info method throws you both datatypes and null values by columns. 
```Python
null_cols = cleanFireDf.isnull().sum()
null_cols[null_cols > 0]
```
- When using describe try to apply on pre-selected columns. It does not make sense to include "ID" types in describe method. 
- Doubt1: is it worth in any case converting categorical data into numerical data (assigning a number to each category) in order to see correlations? I am not sure how accurate this would be.. I tried doing this with the NFFL Fuel Model but correlation very low with all other columns when I expected to have much higher correlation, like for instance FuelType vs Acreage, since fire in a specific fuel would spread much quicker than in others (fallen woord vs green trees). No, a categorical is always a categorical variable even if it's encoded. ML algorithms usually need all categorical variables be encoded.